### DISCUSSION TITLE
StateCreater Mutate does not satisfy constraint [never, unkown]

### DISCUSSION THREAD

**Discussion by Enakers** [tauri-apps/tauri#6677] [Q&A]
Hi,

Trying to use StateCreator with slices but have typescript errors.

`type GeneralSlice = StateCreator<State, [["zustand/immer", never]], [], State>;`

gives the error `typescript: Type '[["zustand/immer", never]]' does not satisfy the constraint '[never, unknown][]'.
  Type '["zustand/immer", never]' is not assignable to type '[never, unknown]'.
    Type at position 0 in source is not compatible with type at position 0 in target.
      Type 'string' is not assignable to type 'never'. [2344]`

Can anyone let me know what I have done wrong?

cheers.

**Comment by soulbliss** [comment]
I am also having the same issue

**Comment by lightDer** [comment]
try `import { immer } from 'zustand/middleware/immer';`.

**Reply by sunny-mittal** [reply]
this works but why it required? This means we have to suppress unused import warnings everywhere (if we build slices in separate files)

**Reply by bootanimation** [reply]
![image](https://github.com/tauri-apps/tauri/assets/161667748/d12e5049-8dea-4069-b83c-22f3de1695b1)
This seems to keep everyone happy :D
